lightning/src/routing/network_graph.rs

```diff
@@ -1765,4 +2154,647 @@ mod tests {
+			assert_eq!(result.is_err(), true);
+		}
+
+		// When no active query exists for a different node, it should send a query message
+		{
+			let result = net_graph_msg_handler.query_channel_range(&node_id_2, chain_hash, first_blocknum, number_of_blocks);
+			assert_eq!(result.is_ok(), true);
+
+			// It should create a task for the query
+			assert!(net_graph_msg_handler.query_channel_range_tasks.lock().unwrap().contains_key(&node_id_2));
+
+			// It should send a query_channel_message with the correct information
+			let events = net_graph_msg_handler.get_and_clear_pending_msg_events();
+			assert_eq!(events.len(), 1);
+			match &events[0] {
+				MessageSendEvent::SendChannelRangeQuery{ node_id, msg } => {
+					assert_eq!(node_id, &node_id_2);
+					assert_eq!(msg.chain_hash, chain_hash);
+					assert_eq!(msg.first_blocknum, first_blocknum);
+					assert_eq!(msg.number_of_blocks, number_of_blocks);
+				},
+				_ => panic!("Expected MessageSendEvent::SendChannelRangeQuery")
+			};
+		}
+	}
+
+	#[test]
+	fn sending_query_short_channel_ids() {
+		let (secp_ctx, net_graph_msg_handler) = create_net_graph_msg_handler();
+		let node_privkey_1 = &SecretKey::from_slice(&[42; 32]).unwrap();
+		let node_id_1 = PublicKey::from_secret_key(&secp_ctx, node_privkey_1);
+
+		let chain_hash = genesis_block(Network::Testnet).header.block_hash();
+
+		// The first query should send the batch of scids to the peer
+		{
+			let short_channel_ids: Vec<u64> = vec![0, 1, 2];
+			let result = net_graph_msg_handler.query_short_channel_ids(&node_id_1, chain_hash, short_channel_ids.clone());
+			assert!(result.is_ok());
+
+			// Validate that we have enqueued a send message event and that it contains the correct information
+			let events = net_graph_msg_handler.get_and_clear_pending_msg_events();
+			assert_eq!(events.len(), 1);
+			match &events[0] {
+				MessageSendEvent::SendChannelsQuery{ node_id, msg } => {
+					assert_eq!(node_id, &node_id_1);
+					assert_eq!(msg.chain_hash, chain_hash);
+					assert_eq!(msg.short_channel_ids, short_channel_ids);
+				},
+				_ => panic!("Expected MessageSendEvent::SendChannelsQuery")
+			};
+		}
+
+		// Subsequent queries for scids should enqueue them to be sent in the next batch which will
+		// be sent when a reply_short_channel_ids_end message is handled.
+		{
+			let result = net_graph_msg_handler.query_short_channel_ids(&node_id_1, chain_hash, vec![0, 1, 2]);
+			assert!(result.is_ok());
+
+			// Validate that we have not enqueued another send message event yet
+			let events = net_graph_msg_handler.get_and_clear_pending_msg_events();
+			assert_eq!(events.len(), 0);
+		}
+	}
+
+	#[test]
+	fn handling_reply_channel_range() {
+		let (secp_ctx, net_graph_msg_handler) = create_net_graph_msg_handler();
+		let node_privkey_1 = &SecretKey::from_slice(&[42; 32]).unwrap();
+		let node_id_1 = PublicKey::from_secret_key(&secp_ctx, node_privkey_1);
+
+		let chain_hash = genesis_block(Network::Testnet).header.block_hash();
```

Comment:
You initiate your task chain hash with `Network::Testnet` but further down you sound to test your error handling with `Network::Bitcoin`, thus likely not hitting the right error as expected.

Maybe you should assert that error string is matching the expected one, even if it's prone to break that's a worthy hardening.